src/incubator/TransitionAnimator/index.tsx

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, {PropsWithChildren, useEffect, useCallback, useImperativeHandle} from 'react';
+import {View as RNView, LayoutChangeEvent} from 'react-native';
+import Animated, {useSharedValue, useAnimatedStyle, withSpring, withTiming, runOnJS} from 'react-native-reanimated';
+import View, {ViewProps} from '../../components/view';
+import {forwardRef, ForwardRefInjectedProps} from '../../commons/new';
+import useHiddenLocations, {HiddenLocation} from './useHiddenLocations';
+const AnimatedView = Animated.createAnimatedComponent(View);
+export {HiddenLocation as TransitionLocation};
+
+const DEFAULT_ANIMATION_VELOCITY = 300;
+const DEFAULT_ANIMATION_CONFIG = {velocity: DEFAULT_ANIMATION_VELOCITY, damping: 18, stiffness: 300, mass: 0.4};
+
+export type TransitionAnimationEndType = 'in' | 'out';
+
+export interface TransitionAnimatorProps extends ViewProps {
```

Comment:
The visibility is decided by the user.
It is working with `Dialog` when using a `Modal`; I'm not sure how exactly it will work with something like a `Toast`, but I think it's possible to change the `enterFrom`; would you like me to test it or should we do that later?

Reply:
No, let's keep it as it is and later on we'll see if it's really a problem